kai/lib.go

```diff
@@ -92,25 +92,7 @@ func GetInventoryReport(cfg *config.Application) (inventory.Report, error) {
 	close(queue)
 
 	// get pods from namespaces using a worker pool pattern
-	for i := 0; i < cfg.Kubernetes.WorkerPoolSize; i++ {
-		go func() {
-			// each worker needs its own clientset
-			clientset, err := client.GetClientSet(kubeconfig)
-			if err != nil {
-				ch.errors <- err
-				return
-			}
-
-			for namespace := range queue {
-				select {
-				case <-ch.stopper:
-					return
-				default:
-					fetchPodsInNamespace(clientset, cfg.Kubernetes, namespace, ch)
-				}
-			}
-		}()
-	}
+	getPodsFromNameSpace(cfg, kubeconfig, ch, queue)
```

Comment:
Love it, good idea to break this out. One nitpick - the function name is very similar to `fetchPodsInNamespace`, what do you think about making it `launchPodWorkerPool` or something like that?